fixed: #169 GPD. BPMN. Colors and fonts.

SDNj opened the pull request.

SDNj committed:
fixed: #169 GPD. BPMN. Colors and fonts.

dofs197 commented:
Посмотрел.
Для режима частичной прорисовки настройки рисования статической картинки задаются в редакторе, а настройки для обведения выполненного процесса - на сервере. Мне кажется это нормальным.
Режим полной прорисовки я запустил и увидел проблемы (не применились стили), но детально не разбирался по причинам:
- этот тип отрисовки сейчас не используется у клиентов
- для корректной портабельности стилей их потребуется включать в БП (внутри PAR файла)

 Замечания: 
- содержимое настройки Общие -> BPMN перенести в Языки -> BPMN (можно заодно перенести выбор языка по умолчанию в Языки)
- настройки по умолчанию должны быть такими как были раньше (чтобы при внедрении поведение по умолчанию не отличалось от текущего), возможно в связи с этим придётся ввести доп. цвета для элементов, тогда настройки лучше всего в них и разместить 
- при нажатии на кнопку "Применить" текущие графы должны обновляться соответственно настройкам (иначе долго придётся стиль подбирать)
- взаимодействие с сервером убрать, т.к. полноценного всё равно нет (синхронизация только в одну сторону, на сервер), режим полной отрисовки в текущей реализации предлагаю не поддерживать

SDNj commented:
>

SDNj commented:
> для корректной портабельности стилей их потребуется включать в БП (внутри PAR файла)

Обсуждали. Различные цветовые настройки в каждом БП может запутать клиента.

> настройки по умолчанию должны быть такими как были раньше (чтобы при внедрении поведение по умолчанию не отличалось от текущего), возможно в связи с этим придётся ввести доп. цвета для элементов, тогда настройки лучше всего в них и разместить

Restore defaults вернет настройки по умолчанию. Или это о другом?

> взаимодействие с сервером убрать, т.к. полноценного всё равно нет (синхронизация только в одну сторону, на сервер), режим полной отрисовки в текущей реализации предлагаю не поддерживать

Можно добавить кнопку Читать с сервера.
Или для полноценной синхронизации добавить скрытое свойство ВремяМодификации и обновлять его на самом сервере тоже.

dofs197 commented:
> > для корректной портабельности стилей их потребуется включать в БП (внутри PAR файла)

> Обсуждали. Различные цветовые настройки в каждом БП может запутать клиента.

Поэтому предлагаю сейчас режим полной отрисовки не поддерживать.

> > настройки по умолчанию должны быть такими как были раньше (чтобы при внедрении поведение по умолчанию не отличалось от текущего), возможно в связи с этим придётся ввести доп. цвета для элементов, тогда настройки лучше всего в них и разместить

> Restore defaults вернет настройки по умолчанию. Или это о другом?

О другом. Нужно вернуть цветовую гамму по умолчанию как была в версии 4.2.

> > взаимодействие с сервером убрать, т.к. полноценного всё равно нет (синхронизация только в одну сторону, на сервер), режим полной отрисовки в текущей реализации предлагаю не поддерживать

> Можно добавить кнопку Читать с сервера. Или для полноценной синхронизации добавить скрытое свойство ВремяМодификации и обновлять его на самом сервере тоже.

Предлагаю не тратить сейчас усилия на это, т.к. с режимом полной отрисовки всё равно не выходит хорошего решения, да и не критично это.

SDNj committed:
сделано:
содержимое настройки Общие -> BPMN перенести в Языки -> BPMN (можно
заодно перенести выбор языка по умолчанию в Языки);
взаимодействие с сервером убрать.

SDNj committed:
сделано: настройки по умолчанию должны быть такими как были раньше

SDNj committed:
support Apply New Styles

SDNj commented:
все реализовано

dofs197 commented:
> настройки по умолчанию должны быть такими как были раньше (чтобы при внедрении поведение по умолчанию не отличалось от текущего), возможно в связи с этим придётся ввести доп. цвета для элементов, тогда настройки лучше всего в них и разместить

Вижу различия по по шрифту и фону заданий. Можно сделать совсем идентично?
![screenshot from 2016-07-30 12 38 22](https://cloud.githubusercontent.com/assets/12091818/17269502/ccbe33e2-5652-11e6-9a7c-6570558ef11e.png)
![screenshot from 2016-07-30 12 39 02](https://cloud.githubusercontent.com/assets/12091818/17269503/ccde29ae-5652-11e6-90d6-4d5c1334993f.png)

SDNj committed:
Feature #167 fixed font and annotation line

dofs197 commented:
#77
